extension/scripts/features/crimes2-burglary-filter/ttBurglaryFilter.js

```diff
@@ -0,0 +1,201 @@
+		});
+	}
+
+	const localFilters = {};
+	async function createFilter(crimeRoot) {
+		document.body.classList.add("torntools-burglary-filter");
+
+		const { content, options } = createContainer("Burglary Filter", {
+			class: "mb10",
+			nextElement: crimeRoot,
+			filter: true,
+		});
+
+		const statistics = createStatistics("targets");
+		content.appendChild(statistics.element);
+		localFilters["Statistics"] = { updateStatistics: statistics.updateStatistics };
+
+		const filterContent = document.newElement({
+			type: "div",
+			class: "content",
+		});
+
+		const targetNameFilter = createFilterSection({
+			title: "Target Name",
+			text: "text",
+			default: filters.burglary.targetName,
+			callback: filtering,
+		});
+		filterContent.appendChild(targetNameFilter.element);
+		localFilters["Target Name"] = { getValue: targetNameFilter.getValue };
+
+		const targetTypeFilter = createFilterSection({
+			type: "Target Type",
+			checkboxes: [
+				{ description: "Residential Targets", id: "residential-targets" },
+				{ description: "Commercial Targets", id: "commercial-targets" },
+				{ description: "Industrial Targets", id: "industrial-targets" },
+			],
+			defaults: filters.burglary.targetType,
+			callback: filtering,
+		});
+		filterContent.appendChild(targetTypeFilter.element);
+		localFilters["Target Type"] = { getSelections: targetTypeFilter.getSelections };
+
+		content.appendChild(filterContent);
+
+		return content;
+	}
+
+	async function addFilter(crimeRoot) {
+		if (!window.location.hash.includes("burglary")) return;
+		else {
+			if (!crimeRoot) {
+				try {
+					crimeRoot = await requireElement(".crime-root.burglary-root");
+				} catch (error) {
+					return;
+				}
+			}
+		}
+
+		if (!findContainer("Burglary Filter")) createFilter(crimeRoot);
+
+		await filtering();
+	}
+
+	async function filtering(argument) {
+		await requireElement(".crime-root.burglary-root [class*='virtualList__'] > [class*='virtualItem__']");
+
+		CRIMES2_ROWS_START_Y = document.find(".virtual-item:first-child")?.style?.height?.getNumber() ?? 64;
+
+		const content = findContainer("Burglary Filter").find("main");
+		const targetName = localFilters["Target Name"].getValue(content).trim();
+		const targetType = localFilters["Target Type"].getSelections(content);
+
+		// Save filters
+		await ttStorage.change({
+			filters: { burglary: { targetName, targetType } },
+		});
+
+		// Actual Filtering
+		// Burglary targets are absolutely positioned on the page, using translateY style.
+		// Changing translateY ourselves to remove holes in targets list. This also preserves Torn's animation.
+		let targetRowHeightsSum = CRIMES2_ROWS_START_Y;
+		for (const targetEl of document.findAll(".crime-root.burglary-root [class*='virtualList__'] > [class*='virtualItem__']:not(:first-child)")) {
+			const rowTargetName = targetEl.find("[class*='crimeOptionSection__']").textContent,
+				rowTargetType = TARGET_TYPE_MAPPING[rowTargetName].toLowerCase().replace(" ", "-");
```

Comment:
Instead of hardcoding these mappings, I would suggest using the image path.

/images/v2/crimes/07-burglary/target-images/**industrial**/webp/x1/burglary_industrial_paper_factory_x1.webp
/images/v2/crimes/07-burglary/target-images/**commercial**/webp/x1/burglary_commercial_chiropractor_x1.webp
/images/v2/crimes/07-burglary/target-images/**residential**/webp/x1/burglary_residential_farmhouse_x1.webp

Reply:
If not, include a safety for targets that we don't have in our list.

It should also be compatible with the 'Crime Morale' script, which it currently is not as `textContent` return "Farmhouse71h". Instead do `.innerText.split("\n")[0]` as that will detect the name completely. Not an issue for the text search though as that works with an includes.

Reply:
Changed them to be extracted from image's URL.